sdktests/server_side_persistence_daemon.go

```diff
@@ -0,0 +1,184 @@
+func (s *ServerSidePersistenceTests) doDaemonModeTests(t *ldtest.T) {
+	t.Run("ignores database initialization flag", s.ignoresInitialization)
+	t.Run("can disable cache", s.canDisableCache)
+	t.Run("caches flag for duration", s.cachesFlagForDuration)
+	t.Run("caches flag forever", s.cachesFlagForever)
+}
+
+func (s *ServerSidePersistenceTests) ignoresInitialization(t *ldtest.T) {
+	persistence := NewPersistence()
+	persistence.SetStore(servicedef.SDKConfigPersistenceStore{
+		Type: servicedef.Redis,
+		DSN:  s.persistence.DSN(),
+	})
+	persistence.SetCache(servicedef.SDKConfigPersistenceCache{
+		Mode: servicedef.Off,
+	})
+	context := ldcontext.New("user-key")
+
+	require.NoError(t, s.persistence.Reset())
+	client := NewSDKClient(t, persistence)
+
+	h.RequireEventually(t, func() bool {
+		result := client.EvaluateFlag(t, servicedef.EvaluateFlagParams{
+			FlagKey:      "flag-key",
+			Context:      o.Some(context),
+			ValueType:    servicedef.ValueTypeAny,
+			DefaultValue: ldvalue.String("default"),
+			Detail:       true,
+		})
+
+		return result.Reason.IsDefined() &&
+			result.Reason.Value().GetErrorKind() == ldreason.EvalErrorFlagNotFound
+	}, time.Second, time.Millisecond*20, "flag was found before it should have been")
+
+	require.NoError(t, s.persistence.WriteData("launchdarkly:features", s.initialFlags))
+	pollUntilFlagValueUpdated(t, client, "flag-key", context,
+		ldvalue.String("default"), ldvalue.String("fallthrough"), ldvalue.String("default"))
+}
+
+func (s *ServerSidePersistenceTests) canDisableCache(t *ldtest.T) {
+	require.NoError(t, s.persistence.Reset())
+	require.NoError(t, s.persistence.WriteData("launchdarkly:features", s.initialFlags))
+
+	persistence := NewPersistence()
+	persistence.SetStore(servicedef.SDKConfigPersistenceStore{
+		Type: servicedef.Redis,
+		DSN:  s.persistence.DSN(),
+	})
+	persistence.SetCache(servicedef.SDKConfigPersistenceCache{
+		Mode: servicedef.Off,
+	})
+
+	context := ldcontext.New("user-key")
+
+	client := NewSDKClient(t, persistence)
+	pollUntilFlagValueUpdated(t, client, "flag-key", context,
+		ldvalue.String("default"), ldvalue.String("fallthrough"), ldvalue.String("default"))
+
+	// Completely reset the database so there are no valid flag definitions
+	require.NoError(t, s.persistence.Reset())
+
+	h.RequireEventually(t,
+		checkForUpdatedValue(t, client, "flag-key", context,
+			ldvalue.String("fallthrough"), ldvalue.String("default"), ldvalue.String("default")),
+		time.Second, time.Millisecond*20, "flag value was NOT updated after cache TTL")
+}
+
+func (s *ServerSidePersistenceTests) cachesFlagForDuration(t *ldtest.T) {
+	persistence := NewPersistence()
+	persistence.SetStore(servicedef.SDKConfigPersistenceStore{
+		Type: servicedef.Redis,
+		DSN:  s.persistence.DSN(),
+	})
+	persistence.SetCache(servicedef.SDKConfigPersistenceCache{
+		Mode: servicedef.TTL,
+		TTL:  o.Some(1),
+	})
+	context := ldcontext.New("user-key")
+
+	t.Run("will cache found flag for TTL", func(t *ldtest.T) {
```

Comment:
I get it now, but the test name is confusing at first - ".. so, will the cache found this flag..?" 

Although once I got it, I couldn't "unsee" it without trying hard.

How about:
> cache hit persists for TTL

> cache miss persists for TTL